crates/autopilot/src/infra/solvers/authenticator.rs

```diff
@@ -0,0 +1,91 @@
+use {
+    crate::{
+        domain::{self, eth},
+        infra::blockchain::{
+            contracts::{deployment_address, Contracts},
+            ChainId,
+        },
+    },
+    ethcontract::{dyns::DynWeb3, GasPrice},
+};
+
+#[allow(dead_code)]
+#[derive(Debug, Clone)]
+pub struct Authenticator {
+    /// The authenticator contract that decides which solver is allowed to
+    /// submit settlements.
+    authenticator: contracts::GPv2AllowListAuthentication,
+    /// The safe module that is used to provide special role to EOA.
+    authenticator_role: contracts::Roles,
+    /// The EOA that is allowed to remove solvers.
+    authenticator_eoa: ethcontract::Account,
+}
+
+///  Authenticator specific addresses
+pub struct Addresses {
+    pub authenticator_eoa: eth::H256,
+}
+
+impl Authenticator {
+    /// Creates an authenticator which can remove solvers from the allow-list
+    pub async fn new(
+        web3: DynWeb3,
+        chain: ChainId,
+        contracts: Contracts,
+        addresses: Addresses,
+    ) -> Self {
+        let authenticator_role = contracts::Roles::at(
+            &web3,
+            deployment_address(contracts::Roles::raw_contract(), &chain).expect("roles address"),
+        );
+
+        Self {
+            authenticator: contracts.authenticator().clone(),
+            authenticator_role,
+            authenticator_eoa: ethcontract::Account::Offline(
+                ethcontract::PrivateKey::from_raw(addresses.authenticator_eoa.0).unwrap(),
+                None,
+            ),
+        }
+    }
+
+    /// Fire and forget: Removes solver from the allow-list in the authenticator
+    /// contract. This solver will no longer be able to settle.
+    #[allow(dead_code)]
+    fn remove_solver(&self, solver: domain::eth::Address) {
+        let calldata = self
+            .authenticator
+            .methods()
+            .remove_solver(solver.into())
+            .tx
+            .data
+            .expect("missing calldata");
+        let authenticator_eoa = self.authenticator_eoa.clone();
+        let authenticator_address = self.authenticator.address();
+        let authenticator_role = self.authenticator_role.clone();
+        let mut byte_array = [0u8; 32];
```

Comment:
nit: hardcoded stuff that can be created in the task should probably also be created in the task instead of moved into it.
Also can we have some documentation (e.g. link) on what this value means or where it comes from?

Reply:
@Mateo-mro this comes from the output of the manual tx we need to do on each network - so, tx where we set the role, then the role number gets assigned.